### ISSUE TITLE
[Feature] Add Severice Endpoint Discovery mechnism

### ISSUE THREAD

**Issue by wonderflow** [kubevela/kubevela#4605]
**Is your feature request related to a problem? Please describe.**

Currently we have `vela status --endpoint` to check all Kubernetes Service/Ingress, this is really useful. But some scenarios especially for SaaS and Cloud Services, they're not always work as Kubernetes Service/Ingress.

For example: 

* Netlify SaaS, we already have an addon: https://github.com/kubevela/catalog/tree/master/addons/netlify . But we can not get the endpoint from `vela status --endpoint`, we can get that information from `vela logs`. We don't even have a mechanism to integrate.

There's also an feature request exist in OAM Spec which is similar to the feature needed here: https://github.com/oam-dev/spec/issues/422

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by oeular** [comment]
Exactly, Add Service Endpoint Discovery mechnism (a bit spelling error, ^_^)

**Comment by wonderflow** [comment]
Need to check this idea:

1. create an empty ingress with the URL inside and mark it as sub-resource in the topology.
2. use vela status --endpoint to get the ingress object

\cc @barnettZQG

**Comment by briankane** [comment]
This issue has been inactive for over 2 years and is scheduled for closure as part of our backlog maintenance.

  **If this issue is still relevant:**
  - Please comment below to keep it open
  - Provide updated context if the original description is outdated

  **Timeline:** If there's no activity by **Friday, October 24th**, this issue will be automatically closed.

  **Note:** Issues can always be reopened if they become relevant again. We encourage the community to reopen and reprioritize any incorrectly 
  closed issues.

**Comment by briankane** [comment]
This issue has been closed following the stale issue notice. 

If this issue is still relevant and should remain open, please:
1. Reopen the issue
2. Remove the 'stale' label
3. Provide an update on the current status or importance

Thank you for your understanding and contribution to KubeVela!
